packages/fast-usdc/src/util/file.js

```diff
@@ -0,0 +1,30 @@
+import { dirname } from 'path';
+
+/** @typedef {import('node:fs/promises').readFile} readAsync */
+/** @typedef {import('node:fs/promises').writeFile} writeAsync */
+/** @typedef {import('node:fs').mkdirSync} mkdirSync */
+/** @typedef {import('node:fs').existsSync} existsSync */
+
+export const makeFile = (
+  /** @type {string} */ path,
+  /** @type {readAsync} */ readFile,
+  /** @type {writeAsync} */ writeFile,
+  /** @type {mkdirSync} */ mkdir,
+  /** @type {existsSync} */ pathExists,
+) => {
```

Comment:
**suggestion**: With more than 3 args, naming them tends to help. It works nicely with `Pick<typeof fs, 'readFile' | 'writeFile' | ...>` too.